markets/perps-market/contracts/modules/PerpsAccountModule.sol

```diff
@@ -54,6 +54,11 @@ contract PerpsAccountModule is IAccountModule {
             account.id = accountId;
         }
 
+        // Check if there are pending orders
+        if (account.hasPendingOrders) {
```

Comment:
Should we add a check here that `amountDelta < 0`? I don't think a user should ever be prevented from depositing collateral, but it makes sense to prevent withdrawals.

Reply:
How could a person manipulate the timing by adding more collateral? I see how they could withdraw to cause the execution to revert, then deposit again when they want to execute.

Reply:
Yea troy's point does make sense to me

Reply:
> How could a person manipulate the timing by adding more collateral? I see how they could withdraw to cause the execution to revert, then deposit again when they want to execute.

it just gives the user more tools to play with, like imagine with me during crashing market prices, the user submits intention at the brink of what is possible with his margin, that by the time the order is elegible for execution, he doesn't have enough margin to get executed, he gets a free option to add collateral to be allowed to execute the pending order during that execution time window

Reply:
It is a safer design therefore, to just allow modification of collateral if no pending order is there

Reply:
so, basically (we had this discussion on perpsV2, and there are some complex vectors or side effects) we want to prevent any change of collateral while orders are pending.

Reply:
@Tburm yeah the attacker would need to check for a favorable with respect to the more lively prices offchain (which are expected to impact his position post-execution)

Reply:
Makes sense, thanks for the details. We should just make sure frontends are well aware of this, since users should know if/when they are prevented from depositing to fix underwater positions.